sdk/storage/Azure.Storage.Blobs/src/BlobClientSpecializedExtensions.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Cryptography;
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Cryptography.Models;
+
+namespace Azure.Storage.Blobs.Specialized
+{
+    /// <summary>
+    /// Specialized extensions for <see cref="BlobClient"/>.
+    /// </summary>
+    public static class BlobClientSpecializedExtensions
+    {
+        /// <summary>
+        /// Rotates the Key Encryption Key (KEK) for a client-side encrypted blob without
+        /// needing to reupload the entire blob.
+        /// </summary>
+        /// <param name="client">
+        /// Client to the blob.
+        /// </param>
+        /// <param name="requestConditions">
+        /// Request conditions for accessing the blob.
+        /// </param>
+        /// <param name="newKeyOverride">
+        /// Optional override of the new Key Encryption Key for the blob. Will default to the
+        /// client's <see cref="ClientSideEncryptionOptions.KeyEncryptionKey"/>.
+        /// </param>
+        /// <param name="oldKeyResolverOverride">
+        /// Optional override for the resolver to resolve the blob's current Key Encryption
+        /// Key. Will default to the client's <see cref="ClientSideEncryptionOptions.KeyResolver"/>.
+        /// </param>
+        /// <param name="keywrapAlgorithmOverride">
+        /// Optional override for the key-wrap algorithm to re-wrap the Content Encryption Key.
+        /// Will default to the client's <see cref="ClientSideEncryptionOptions.KeyWrapAlgorithm"/>.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Cancellation token for the operation.
+        /// </param>
+        public static void RotateClientSideEncryptionKey(
+            this BlobClient client,
+            BlobRequestConditions requestConditions = default,
+            IKeyEncryptionKey newKeyOverride = default,
+            IKeyEncryptionKeyResolver oldKeyResolverOverride = default,
+            string keywrapAlgorithmOverride = default,
+            CancellationToken cancellationToken = default)
```

Comment:
@tg-msft Can we get your opinion on this? I think I'm leaning towards `RotateClientSideEncryptionKeyOptions` as described above where you either use the all of the client-wide `ClientSideEncryptionOptions` or all of the method provided `ClientSideEncryptionOptions` with no mixing of the two. Argument validation for key rotation would be as follows:
```csharp
ClientSideEncryptionOptions optionsToUseForRotation = options?.EncryptionOptionsOverrides
    ?? client.ClientSideEncryption
    ?? throw new ArgumentException("some message");
```

Reply:
I also prefer `RotateClientSideEncryptionKeyOptions`

Reply:
I'd suggest the verb `Update` instead of `Rotate` because that's what we're using everywhere for rolling credentials like https://apiview.dev/Assemblies/Review/77bdbcef49a3465fa96fac3fb745bf57#Azure.AzureKeyCredential.  _(`StorageSharedKeyCredential.SetAccountKey` predates that pattern and we'd have called it `UpdateAccountKey` if writing it today.)_